### PR TITLE
fix(CustomFrame): Resolves issue #21731 where date range in explore throws runtime error

### PR DESCRIPTION
**Pull request by eric-briscoe**
### SUMMARY

Resolves issue #21731 where date range in explore throws runtime error when explore.common is undefined in Redux state.  
See associated issue: https://github.com/apache/superset/issues/21731

This issue is reproducible from the 2.0.0 and 2.0.1 RC1 releases but the change also improves runtime resilience of master.  PR is targeting master, but intended to be cherry picked into the 2.0.1 RC1 as well.

Changes made in superset/superset-frontend/src/explore/components/controls/DateFilterControl/components/CustomFrame.tsx 

in PR https://github.com/apache/superset/pull/20063/files#diff-ac08ce133d14a8a92607bf02bc368b6b52b8314ddef1b872e42746ba8d6038d0

introduced a bug that expects redux state to contain a path that may be undefined, and assume that a provided locale is valid and known by antd.  This fix ensures safety checks are made and that antd DatePicker can fall back to internal default is a compatible locale is not identified.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

See associated issue: https://github.com/apache/superset/issues/21731

### TESTING INSTRUCTIONS

See associated issue: https://github.com/apache/superset/issues/21731

### ADDITIONAL INFORMATION

- [X] Has associated issue: #21731 
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/controls/DateFilterControl/components/CustomFrame.test.tsx [hunk 1]
```diff
@@ -39,6 +39,14 @@ const store = mockStore({
   common: { locale: 'en' },
 });
 
+// case when common.locale is not populated
+const mockEmptyStore = configureStore([thunk]);
+const emptyStore = mockEmptyStore({});
+
+// case when common.locale is populated with invalid locale
+const mockInvalidStore = configureStore([thunk]);
+const invalidStore = mockInvalidStore({ common: { locale: 'invalid_locale' } });
```
Comment (eschutho) [inline superset-frontend/src/explore/components/controls/DateFilterControl/components/CustomFrame.test.tsx:48]: I'm not familiar with the return object of configureStore but can you create two separate distinct stores with the same return of `configureStore`? In other words, can we create empty and invalid both from the `mockEmptyStore`?

**Review by eschutho — APPROVED** [review]
just left one question/possible nit about duplication of the test store.

File: superset-frontend/src/explore/components/controls/DateFilterControl/components/CustomFrame.test.tsx (hunk 1 above)
Comment (eric-briscoe) [reply superset-frontend/src/explore/components/controls/DateFilterControl/components/CustomFrame.test.tsx]: Great catch!  I was able to consolidate one call to configureStore, and reuse calls to mockStore with the different values.

**Review by zhaoyongjie — APPROVED** [review]
thanks for the fix and add new test case!

**Pull request merged**